packages/react-composites/src/composites/CallComposite/components/LocalPreview.tsx

```diff
@@ -149,7 +149,7 @@ export const LocalPreview = (props: LocalPreviewProps): JSX.Element => {
               data-ui-id="call-composite-local-device-settings-options-button"
               {...devicesButtonProps}
               // disable button whilst all other buttons are disabled
-              disabled={!microphonePermissionGranted || !cameraPermissionGranted || hasNoDevices}
```

Comment:
I suppose technically we should be checking both the permission cross referencing those if there are devices for each?

```ts
const hasNoMics = !microphonePermissionGranted || !devicesButtonProps.microphones.length;
const hasNoSpeakers = !devicesButtonProps.speakers.length; // any permission needed here?
const hasNoCamera = !cameraPermissionGranted || !devicesButtonProps.cameras.length;
disabled = hasNoMics && hasNoSpeakers && hasNoCamera;
```

Reply:
Thinking of the case where we have cameras, but no camera permission

Reply:
Although no permission maybe means we have no cameras... Though not sure if no camera via the role capability means that

Reply:
Actually yeah no we just need to check the camera permission - ignore all this ^_^

Reply:
Reopening -- but only because we might have no camera permission and no mic permission -- but still want to change the speaker.

Reply:
> Thinking of the case where we have cameras, but no camera permission

No! I was thinking of the case where we have camera permission, but no cameras _and_ we have microphones but no microphone permission (and lets assume no speakers). Then the button won't be disabled when it should be

Reply:
Can you tell its the Friday before xmas 😄

Reply:
I like this, as it does a check for when devices have no microphone or camera present despite having permissions.
However, the hasNoDevices does something similar here for this specific case so we are covered I believe.

Reply:
Looks like there's still a couple edge cases:
- no camera or microphone permission but user wishes to change speaker (button will be disabled when it should not)
- we have camera permission and no cameras, and we have no microphone permission but do have mics, and have no speakers (button will be enabled when it should not)